plugins/providers/docker/provider.rb

```diff
@@ -146,18 +146,26 @@ def ssh_info
         # If the container isn't running, we can't SSH into it
         return nil if state.id != :running
 
-        port_name = "#{@machine.config.ssh.guest_port}/tcp"
+        guest_port = @machine.config.ssh.guest_port
+        port_name = "#{guest_port}/tcp"
         network   = driver.inspect_container(@machine.id)['NetworkSettings']
 
-        if network["Ports"][port_name].respond_to?(:first)
+        if driver.podman?
+          port = network["Ports"].select {|p| p["protocol"] == "tcp" and p["containerPort"] == guest_port } .first
```

Comment:
You can drop the `.first` by using a `.detect` instead of `.select` here:

```ruby
port = network["Ports"].detect {|p| p["protocol"] == "tcp" and p["containerPort"] == guest_port }
```

Reply:
amazing!